### PR TITLE
Implement followers logic

### PR DESCRIPTION
**Pull request by mateo762**
Implement the logic to follow a user and to receive a follow from a user, created two lists in UserEntity, "followersUsers" and "followingUsers", also implemented that through the search fragment you can click on a username and it takes you to their profile, in there you can decide if you want to follow them, and if you already follow them you can unfollow them.

Screenshot 2023-04-27 123917
Screenshot 2023-04-27 123926

### CODE REVIEW

**Review by j-encinas — COMMENTED** [review]
Good work on this task! Especially considering you needed to make quite a few updates to previous code aside from the new feature itself. It is good that you also already included tests for this feature, which is appreciated.

**Review by j-encinas — APPROVED** [review]

**Comment by j-encinas** [comment]
For the refinement of the UI component, I would leave it for a later task that unifies the look and appearance of the recently added features in a batch to save time

File: app/src/main/java/com/github/mateo762/myapplication/profile/ProfileActivity.kt
[the remark this answers is not in the document]
Comment (mateo762) [reply app/src/main/java/com/github/mateo762/myapplication/profile/ProfileActivity.kt:1]: Thanks for noticing, I have removed them now

File: app/src/main/java/com/github/mateo762/myapplication/ui/upload/UploadImageScreen.kt
[the remark this answers is not in the document]
Comment (mateo762) [reply app/src/main/java/com/github/mateo762/myapplication/ui/upload/UploadImageScreen.kt:1]: Correct!

**Pull request merged**